go/ql/lib/semmle/go/frameworks/XPath.qll

```diff
@@ -142,6 +142,21 @@ module XPath {
       }
     }
 
+    /**
+     * An XPath expression string used in an API function of the
+     * [lestrrat-go/libxml2](https://github.com/lestrrat-go/libxml2) package.
+     */
+    private class LestratGoLibxml2XPathExpressionString extends Range {
+      LestratGoLibxml2XPathExpressionString() {
+        exists(Function f, string name | name.matches("Parse%") |
+          f.hasQualifiedName("github.com/lestrrat-go/libxml2/parser.Parser", name) and
+          //f.hasQualifiedName(package("github.com/lestrrat-go/libxml2/parser.Parser", ""), name) and
```

Comment:
I could use some help here to understand the proper `mod` and `path` for this library.  Everything I have tried is not matching, but a simple has qualified name works fine to detect my test case.

Reply:
Here is an attempt at a helpful guess:
`f.hasQualifiedName("github.com/lestrrat-go/libxml2/parser", "Parser", name)`

Reply:
The way I figure it out is to look at [the relevant go.mod file](https://github.com/lestrrat-go/libxml2/blob/master/go.mod). The first line gives you the module path. Also, if you make it a `Method m` instead of a `Function f` then there's a nice three-argument version of `hasQualifiedName`. So `m.hasQualifiedName(package("github.com/lestrrat-go/libxml2", "parser"), "Parser", "ParseString")`.

Reply:
PERFECT! Thank you that is what I needed there - resolved: https://github.com/github/codeql/pull/15181/commits/4c46be1ed00e5255103a72cd2fcd4fe75395c6d3